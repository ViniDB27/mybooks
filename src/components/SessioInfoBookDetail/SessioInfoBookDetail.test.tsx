import render from "react-test-renderer";

import { SessionInfoBookDetail } from "./index";

const bookMock = {
  kind: "Mock",
  id: "1",
  etag: "Mock",
  selfLink: "Mock",
  volumeInfo: {
    title: "Mock",
    authors: ["Mock"],
    publisher: "Mock",
    publishedDate: "Mock",
    description: "Mock",
    pageCount: 10,
    imageLinks: {
      smallThumbnail: "Mock",
      thumbnail: "Mock",
    },
    language: "Mock",
  },
};

describe("<SessionInfoBookDetail />", () => {
  const tree = render
    .create(
      <SessionInfoBookDetail
        book={bookMock}
        favorite={false}
      />
    )
    .toJSON();

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });
});
