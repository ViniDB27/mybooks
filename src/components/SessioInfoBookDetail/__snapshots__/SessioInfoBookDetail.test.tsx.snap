// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SessionInfoBookDetail /> renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 200,
      "marginBottom": 25,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "Mock",
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
        "paddingLeft": 15,
      }
    }
  >
    <Text>
      Publicadora: 
      Mock
    </Text>
    <Text>
      Autor: 
      Mock
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0AF",
            "borderRadius": 5,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "space-evenly",
            "marginTop": 10,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 15,
              "fontWeight": "bold",
            }
          }
        >
          Favoritar
        </Text>
      </View>
    </View>
  </View>
</View>
`;
